support-frontend/app/controllers/CreateSubscriptionController.scala

```diff
@@ -142,13 +151,57 @@ class CreateSubscriptionController(
     }
   }
 
+  private def validateBenefits(response: UserBenefitsResponse): EitherT[Future, CreateSubscriptionError, Unit] = {
+    if (response.benefits.contains("adFree")) {
+      // Not eligible
+      EitherT.leftT(RequestValidationError("guardian_light_purchase_not_allowed"))
+    } else {
+      // Eligible
+      EitherT.rightT(())
+    }
+  }
+
+  private def validateUserIsEligibleForPurchase(
+      request: Request[CreateSupportWorkersRequest],
+      userDetails: UserDetailsWithSignedInStatus,
+  ): EitherT[Future, CreateSubscriptionError, Unit] = {
+    request.body.product match {
+      case GuardianLight(_) => {
+        if (userDetails.isSignedIn) {
+          // If the user is signed in, we'll assume they're eligible as they shouldn't have got
+          // to this point of the journey.
+          EitherT.rightT(())
+        } else {
+          for {
+            benefits <- userBenefitsApiService
+              .getUserBenefits(userDetails.userDetails.identityId)
+              .leftMap(_ => ServerError("Something went wrong calling the user benefits API"))
+            _ <- validateBenefits(benefits)
+          } yield ()
+        }
+      }
+      case _ => EitherT.rightT(())
+    }
+  }
+
+  case class UserDetailsWithSignedInStatus(
+      userDetails: UserDetails,
+      isSignedIn: Boolean,
+  )
+
   private def createSubscription(implicit
       settings: AllSettings,
       request: CreateRequest,
   ): EitherT[Future, CreateSubscriptionError, CreateSubscriptionResponse] = {
 
     val maybeLoggedInUserDetails =
```

Comment:
Should this now be "maybeUserDetails" since the signed in status can vary?

Reply:
I had this same thought and decided to leave it as is because in the case of this `val` it's still either:

1) `None` or
2) The details of a user who is signed in.

I.e. `maybeLoggedInUserDetails` won't ever contain information about a user who isn't signed in. Though like you say it's true that the `Option` type `UserDetailsWithSignedInStatus` can represent this state. What do you reckon?

Reply:
ah, right. That's quite subtle but seems fine to leave as is